src/libs/ValidationUtils.js

```diff
@@ -17,6 +18,24 @@ function isValidAddress(value) {
     return !CONST.REGEX.PO_BOX.test(value);
 }
 
+/**
+ * Used to validate a value that is "required".
```

Comment:
This comment to me suggests a method that would do this kind of logic

```js
isRequirementFulfilled() {
    if (_.isString(value)) {
        return !_.isEmpty(value.trim());
    }
    if (_.isArray(value) || _.isObject(value)) {
        return !_.isEmpty(value);
    }
    if (_.isBoolean(value)) {
        return value;
    }
    return Boolean(value);
}
```

Reply:
Hmm, that would be assuming `false` is not valid value for a "required" field, right?

I changed this because it is not so obvious that `false` means that the value is missing. You could a field with possible values `false` | `true` | `undefined`, for example, and interface with two radio buttons "Yes" (`true`) and "No" (`false`) and none of them start selected (`undefined`). In this case, `true` and `false` would suffice if you are just validating "required". But then, this is not really a case we have right now, so an option is to stick to the previous way of the function and think about it if we get to the case where this can be a problem:

```
isRequirementFulfilled() {
    if (_.isString(value)) {
        return !_.isEmpty(value.trim());
    }
    if (_.isArray(value) || _.isObject(value)) {
        return !_.isEmpty(value);
    }
    return Boolean(value);
}
```

About the name of the function `isRequirementFulfilled`, I feel like `requirement` is broad and could be any kind of requirement, not a value being "required" (or mandatory), but then it could be because I'm not a native english speaker :P. Meanwhile, `isRequiredFulfilled` may be better because it is about the "required" requirement.

Reply:
>Hmm, that would be assuming false is not valid value for a "required" field, right?

All of the scenarios we have today suggest a checkbox that is not checked (`false`) should not pass a "required" field constraint. 

>You could a field with possible values false | true | undefined, for example, and interface with two radio buttons

I understand your example, but I would maybe ask that if we are requiring a response to a true/false question then why are we giving the option to input `undefined` at all? Will we ever have a need for something like "boolean, but 'required' means 'true' or 'false', but not 'undefined'"?

>I feel like requirement is broad and could be any kind of requirement, not a value being "required" (or mandatory), but then it could be because I'm not a native english speaker

Makes sense to me! I like the suggestion to make this `isRequiredFulfilled`.

Reply:
> > Hmm, that would be assuming false is not valid value for a "required" field, right?
> 
> All of the scenarios we have today suggest a checkbox that is not checked (`false`) should not pass a "required" field constraint.
> 
> > You could a field with possible values false | true | undefined, for example, and interface with two radio buttons
> 
> I understand your example, but I would maybe ask that if we are requiring a response to a true/false question then why are we giving the option to input `undefined` at all? Will we ever have a need for something like "boolean, but 'required' means 'true' or 'false', but not 'undefined'"?

I think you are pointing to that I'm overthinking/over-engineering this. If it is that, I agree, because we don't have such cases now and maybe we won't have them. If we stick to what we are dealing with right now, as you say, the `false` value should not pass the "required" field constraint.

If you agree with that, I'll revert to how it was working at first.

Reply:
>If you agree with that, I'll revert to how it was working at first. This would also include putting hasNoConnectionToCannabis back in this.requiredFields.

Yep let's do it and get this merged.